### PR TITLE
Add negative index to PixelAccess

### PR DESCRIPTION
**Pull request by Nazime**
Add a negative indexing to the PixelAccess so we can get the last pixels easily

```python
from PIL import Image

im = Image.open("image.png")
pix = im.load()
print(pix[-1, -1])
```

### CODE REVIEW

File: src/_imaging.c [hunk 1]
```diff
@@ -471,7 +471,12 @@ getpixel(Imaging im, ImagingAccess access, int x, int y)
       INT32 i;
       FLOAT32 f;
     } pixel;
-
+    if (x < 0){
+        x = im->xsize + x;
+    }
+    if (y < 0){
```
Comment (hugovk) [inline src/_imaging.c]: Please add a space between `)` and `{`, here and above.

File: src/PIL/PyAccess.py [hunk 2]
```diff
@@ -88,8 +88,12 @@ def __getitem__(self, xy):
         :returns: a pixel value for single band images, a tuple of
           pixel values for multiband images.
         """
-
-        (x, y) = self.check_xy(xy)
+        (x, y) = xy
+        if (x < 0):
+            x = self.xsize + x
+        if (y < 0):
```
Comment (hugovk) [inline src/PIL/PyAccess.py]: Please remove redundant parentheses, here and above.

File: src/_imaging.c [hunk 3]
```diff
@@ -471,7 +471,12 @@ getpixel(Imaging im, ImagingAccess access, int x, int y)
       INT32 i;
       FLOAT32 f;
     } pixel;
-
+    if (x < 0){
```
Comment (hugovk) [inline src/_imaging.c]: Let's keep that line of whitespace after `pixel;`

File: src/_imaging.c [hunk 4]
```diff
@@ -471,7 +471,12 @@ getpixel(Imaging im, ImagingAccess access, int x, int y)
...
+        x = im->xsize + x;
+    }
+    if (y < 0){
+        y = im->ysize + y;
+    }
```
Comment (hugovk) [inline src/_imaging.c]: And add a line of whitespace after this.

**Comment by radarhere** [comment]
Thanks for this.

You’ve added getting the negative index - it would be good to also add setting.

[`im.load()` returns either PyAccess (Python) or PixelAccess (C)](https://github.com/python-pillow/Pillow/blob/master/src/PIL/Image.py#L842-L843 ). So that’s why you are finding that you need to change the C code.

**Comment by Nazime** [comment]
What do you mean by add setting?

**Comment by radarhere** [comment]
You have added support for `print(pix[-1, -1])`, but not `pix[-1, -1] = (0, 0, 0)` - the first being 'get', the second being 'set'.

**Comment by Nazime** [comment]
Oh yes, I'll add that too.

File: Tests/test_image_access.py [hunk 5]
```diff
@@ -64,6 +64,43 @@ def test_sanity(self):
 
         self.assert_image_equal(im1, im2)
 
+    def test_sanity_negative_index(self):
+        im1 = hopper()
+        im2 = Image.new(im1.mode, im1.size, 0)
+
+        width, heigth = im1.size
```
Comment (radarhere) [inline Tests/test_image_access.py]: height

Comment (Nazime) [reply Tests/test_image_access.py]: Oups didn't see it

**Comment by hugovk** [comment]
Let' remember to add this to the release notes.

**Pull request merged**